src/jvm/main/org/jetbrains/kotlinx/lincheck/bootstrap/Injections.kt

```diff
@@ -0,0 +1,221 @@
+/*
+ * Lincheck
+ *
+ * Copyright (C) 2019 - 2024 JetBrains s.r.o.
+ *
+ * This Source Code Form is subject to the terms of the
+ * Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
+ * with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+// we need to use some "legal" package for the bootstrap class loader
+@file:Suppress("PackageDirectoryMismatch")
+
+package sun.nio.ch.lincheck
+
+import org.jetbrains.kotlinx.lincheck.runInIgnoredSection
+import java.util.*
+
+internal object Injections {
```

Comment:
TODO: needs to be documented. 

We need both a per-method documentation, as well as a documentation giving a high-level summary explanation on how our code instrumentation works.